Scripts/RemediationScripts/Remediate-EnableHTTPSUrlSchemeForAPIManagementServices.ps1

```diff
@@ -0,0 +1,769 @@
+        break
+    }
+
+    Write-Host "Fetching all API Management services from $($FilePath)" -ForegroundColor $([Constants]::MessageType.Info)
+    $validAPIManagementServiceDetails = @()
+    $apiManagementServiceDetails = Import-Csv -LiteralPath $FilePath
+    $apiManagementServiceDetails | ForEach-Object{
+        if( ![String]::IsNullOrWhiteSpace($_.ResourceId) -and ![String]::IsNullOrWhiteSpace($_.ResourceGroupName) -and ![String]::IsNullOrWhiteSpace($_.ResourceName) )
+        {
+            $validAPIManagementServiceDetails += $_
+        }
+    }
+
+    $totalAPIManagementResources = $($validAPIManagementServiceDetails.Count)
+
+    if ($totalAPIManagementResources -eq 0)
+    {
+        Write-Host "No API Management services found. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+        break
+    }
+
+    Write-Host "Found $($totalAPIManagementResources) API Management Service(s)." -ForegroundColor $([Constants]::MessageType.Update)
+    
+    # Back up snapshots to `%LocalApplicationData%'.
+    $backupFolderPath = "$([Environment]::GetFolderPath('LocalApplicationData'))\AzTS\Remediation\Subscriptions\$($context.Subscription.SubscriptionId.replace('-','_'))\$($(Get-Date).ToString('yyyyMMddhhmm'))\DisableHTTPSUrlSchemeForAPIsOfAPIManagementServices"
+
+    if (-not (Test-Path -Path $backupFolderPath))
+    {
+        New-Item -ItemType Directory -Path $backupFolderPath | Out-Null
+    }
+ 
+    Write-Host "HTTPS url scheme will be disabled on the API(s) of all API Management service." -ForegroundColor $([Constants]::MessageType.Warning)
+    
+    if (-not $Force)
+    {
+        Write-Host "Do you want to disable HTTPS url scheme for API(s) of all API Management services?" -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+            
+        $userInput = Read-Host -Prompt "(Y|N)"
+
+        if($userInput -ne "Y")
+        {
+            Write-Host "HTTPS url scheme will not be disabled for API(s) of any API Management Service. Exiting..." -ForegroundColor $([Constants]::MessageType.Update)
+            break
+        }
+    }
+    else
+    {
+        Write-Host "'Force' flag is provided. HTTPS url scheme will be disabled on the API(s) of API Management services without any further prompts." -ForegroundColor $([Constants]::MessageType.Warning) -NoNewline
+    }
+
+    Write-Host $([Constants]::DoubleDashLine)
+    Write-Host "[Step 3 of 3] Disabling HTTPS url scheme for API(s) of all API Management services..." -ForegroundColor $([Constants]::MessageType.Warning)
+
+    # Includes API Management services, to which, previously made changes were successfully rolled back.
+    $apiManagementServicesRolledBack = @()
+
+    # Includes API Management services that were skipped during roll back. There were errors rolling back the changes made previously.
+    $apiManagementServicesSkipped = @()
+
+
+    $validAPIManagementServiceDetails | ForEach-Object {
+        $apiManagementservice = $_
```

Comment:
Please rename the variable from `apiManagementservice` to `apiManagementService`.